examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineSingleUnitExample.cs

```diff
@@ -142,12 +144,23 @@ private static PACalculationParametersRoot GetPaCalculationParameters()
             var paComponentId = componentsResponse.Data.FirstOrDefault(component => (component.Value.Name == ComponentName && component.Value.Category == ComponentCategory)).Key;
             Console.WriteLine($"PA Component Id : {paComponentId}");
 
+            var pricingSourcesApi = new PricingSourcesApi(GetApiConfiguration());
+
+            var pricingSourcesResponse = pricingSourcesApi.GetPAPricingSources(PricingSourceName, PricingSourceCategory, PricingSourceDirectory);
+
+            var paPricingSourceId = pricingSourcesResponse.Data.FirstOrDefault(pricingSource => (pricingSource.Value.Name == PricingSourceName && pricingSource.Value.Category == PricingSourceCategory && pricingSource.Value.Directory == PricingSourceDirectory)).Key;
+            Console.WriteLine($"PA Pricing Source Id : {paPricingSourceId}");
+
             var paAccountIdentifier = new PAIdentifier(Portfolio);
             var paAccounts = new List<PAIdentifier> { paAccountIdentifier };
             var paBenchmarkIdentifier = new PAIdentifier(Benchmark);
             var paBenchmarks = new List<PAIdentifier> { paBenchmarkIdentifier };
 
-            var paCalculation = new PACalculationParameters(paComponentId, paAccounts, paBenchmarks);
+            var paPortfolioPricingSources = new List<PACalculationPricingSource> { new PACalculationPricingSource(id: paPricingSourceId) };
+
+            var paDataSources = new PACalculationDataSources(portfoliopricingsources: paPortfolioPricingSources, useportfoliopricingsourcesforbenchmark:true); 
+
+            var paCalculation = new PACalculationParameters(paComponentId, paAccounts, paBenchmarks, datasources: paDataSources);
```

Comment:
Can you please pass all of them as named arguments?

Reply:
Updated